docs/retention-policy-agent/README.md

```diff
@@ -12,7 +12,75 @@ weight: 2
 The Results Retention Policy Agent removes older Results and their associated Records from the DB.
 
 ## Configuration
-The Results Retention Policy Agent config uses following field
 
-- `runAt`: determines when to run pruning job for DB.
-- `maxRetention`: for how long to keep results and records.
+The Results Retention Policy Agent is configured via the `tekton-results-config-results-retention-policy` ConfigMap.
+
+The following fields are supported:
+
+- `runAt`: Determines when to run the pruning job for the DB. It uses a cron schedule format. The default is `"7 7 * * 7"` (every Sunday at 7:07 AM).
+- `maxRetention`: The default retention period for how long to keep Results and Records when no specific policy matches. This can be a number (e.g., `30`), which is interpreted as days, or a duration string (e.g., `30d`, `24h`). The default is `30d`.
+- `policies`: A list of fine-grained retention policies that allow for more specific control over data retention.
+
+### Fine-Grained Retention Policies
+
+You can define a list of policies to control retention based on various criteria. The `policies` field in the ConfigMap accepts a YAML string containing a list of policy objects. Each policy has a `name`, a `selector`, and a `retention` period.
```

Comment:
Does this need to be a string or could it be an object instead? Strings which are also yaml are poor UX in my opinion, especially if the string is already in a yaml document, since the yaml wont be validated it makes it difficult for CI to validate the settings are valid and easy for a user to introduce yaml errors.

Reply:
It's not a string TBH. It's marshal to a struct.

Reply:
We take a yaml and marshal it to a struct. Isn't that how we do with other configmap?